src/error.rs

```diff
@@ -32,4 +45,56 @@ pub enum DatabaseError {
     WriteError(String),
     #[error("Failed to delete {0} from the Database database")]
     DeleteError(String),
-}
\ No newline at end of file
+}
+
+#[derive(Error, Debug)]
+pub enum DataAvailabilityError {
+    #[error("Failed to establish WebSocket connection: {0}")]
+    WebSocketError(String),
+    #[error("Namespace initialization error")]
+    NamespaceInitializationError,
+    #[error("The data channel has been closed")]
+    ChannelClosed,
+    #[error("Network error: {0}")]
+    NetworkError(String),
+    #[error("Data retrieval error at height {0}: {1}")]
+    DataRetrievalError(u64, String),
+    #[error("Error parsing data at height {0}")]
+    ParsingError(u64),
+    #[error("Error submitting data at height {0}: {1}")]
+    WriteError(u64, String),
```

Comment:
SubmissionError seems more clear to me but your choice

Reply:
I agree!